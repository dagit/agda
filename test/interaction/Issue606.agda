module Issue606 where

infixr 1 _,_

record _×_ (A B : Set) : Set where
  constructor _,_
  field fst : A
        snd : B

postulate A B C : Set

test : A × (B × C)
test = {!!} , {!!}
-- refining the second hole should give "? , ?"  (no enclosing parens!)
