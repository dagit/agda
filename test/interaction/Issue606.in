top_command (cmd_write_highlighting_info currentFile)
top_command (cmd_load currentFile [])
goal_command 1 (cmd_refine_or_intro False) ""

