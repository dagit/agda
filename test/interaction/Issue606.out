Agda2> (agda2-status-action "")
Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue606 (Issue606.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue606.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : A\n?1 : B Ã— C\n" nil)
((last . 1) . (agda2-goals-action '(0 1)))
Agda2> (agda2-give-action 1 "? , ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : A\n?2 : B\n?3 : C\n" nil)
((last . 1) . (agda2-goals-action '(0 2 3)))
Agda2> 